src/js/services/link_track.js

```diff
@@ -0,0 +1,186 @@
+/* eslint-env browser */
+/* global $ */
+const SEND_COMMAND = 'send';
+const EVENT_HIT_TYPE = 'event';
+const EVENT_FIELDS = ['eventCategory', 'eventAction', 'eventLabel', 'eventValue', 'transport', 'hitCallback'];
+
+/**
+ * Compulsory fields for send event on clicking a link.
+ */
+function compulsoryFields(/* link */) {
+  return {
+    transport: 'beacon',
+  };
+}
+
+/**
+ * Returns the URL origin given a string representation of it. If the browser
+ * does not support URL, it returns href itself.
+ *
+ * @param {String} href the string containing the URL.
+ * @return {String} the origin of the URL.
+ */
+function getOrigin(href) {
+  if (URL !== undefined) {
+    return new URL(href).origin;
+  }
+
+  return href;
+}
+
+/**
+ * Computes the default values for some keys to the send event command.
+ *
+ * @param {HTMLAnchorElement} link anchor clicked.
+ * @return {Object} object with the default values.
+ */
+function defaultsForLink(link) {
+  const sameOrigin = getOrigin(link) === window.location.origin;
+  return {
+    eventCategory: sameOrigin ? 'inbound' : 'outbound',
+    eventAction: 'click',
+    eventLabel: link.href,
+  };
+}
+
+/**
+ * Computes the data for a given anchor element
+ *
+ * @param {HTMLAnchorElement} link
+ * @return {Object} object with the values to call ga send event command.
+ */
+function getData(link) {
+  const data = Object.assign(
+    defaultsForLink(link),
+    link.dataset,
+    compulsoryFields(link),
+  );
+
+  return EVENT_FIELDS.reduce((memo, key) => {
+    /* eslint no-param-reassign: ["error", { "props": false }] */
+    if (key in data) {
+      memo[key] = data[key];
+    }
+    return memo;
+  }, {});
+}
+
+/**
+ * Returns the arguments to call ga command when the link has been clicked.
+ *
+ * @param {HTMLAnchorElement} link element to get the data from.
+ * @return {Array<String, String, Object>} arguments for ga method.
+ */
+function commandArguments(link) {
+  return [SEND_COMMAND, EVENT_HIT_TYPE, getData(link)];
+}
+
+const gaInternal = [];
```

Comment:
great! in that case it could be useful if it is documented how can it be used.
Maybe printing its content? what things does it contains? can be pushed events there, and they're sent to GA? is there where the GA events goes when they're pushed to GA, so it can be used to see which events were sent?